src/main/java/org/openstreetmap/atlas/checks/database/FlagDatabaseSubCommand.java

```diff
@@ -0,0 +1,373 @@
+import org.openstreetmap.atlas.utilities.time.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Upload Atlas Checks flags into a Postgres database.
+ *
+ * @author danielbaah
+ */
+public class FlagDatabaseSubCommand extends AbstractAtlasShellToolsCommand
+{
+    private static final String FLAG_PATH_INPUT = "flag_path";
+    private static final String DATABASE_URL_INPUT = "database_url";
+    private static final String LOG_EXTENSION = "log";
+    private static final String ZIPPED_LOG_EXTENSION = ".log.gz";
+    private static final String ISO_COUNTRY_CODE = "iso_country_code";
+    private static final int THREE = 3;
+    private static final int FOUR = 4;
+    private static final int FIVE = 5;
+    private static final int SIX = 6;
+    private static final int SEVEN = 7;
+    private static final int EIGHT = 8;
+    private static final int BATCH_SIZE = 1000;
+
+    /**
+     * An enum containing the different types of input files that we can handle.
+     */
+    private enum OutputFileType
+    {
+        LOG,
+        COMPRESSED_LOG
+    }
+
+    private static final Gson gson = new GsonBuilder()
+            .registerTypeAdapter(CheckFlag.class, new CheckFlagDeserializer()).create();
+    private static final Logger logger = LoggerFactory.getLogger(FlagDatabaseSubCommand.class);
+    private static final Set<String> blacklistKeys = new HashSet<>();
+
+    static
+    {
+        blacklistKeys.add("itemType");
+        blacklistKeys.add("identifier");
+        blacklistKeys.add("osmid");
+        blacklistKeys.add("osmIdentifier");
+        blacklistKeys.add("relations");
+        blacklistKeys.add("members");
+        blacklistKeys.add(ISO_COUNTRY_CODE);
+    }
+
+    private final OptionAndArgumentDelegate optionAndArgumentDelegate;
+    private Timestamp timestamp;
+
+    public static void main(final String[] args)
+    {
+        new FlagDatabaseSubCommand().runSubcommandAndExit(args);
+    }
+
+    public FlagDatabaseSubCommand()
+    {
+        this.optionAndArgumentDelegate = this.getOptionAndArgumentDelegate();
+    }
+
+    @Override
+    @SuppressWarnings("squid:S3655")
+    public int execute()
+    {
+        final Time timer = Time.now();
+        final DatabaseConnection database = new DatabaseConnection(
+                this.optionAndArgumentDelegate.getOptionArgument(DATABASE_URL_INPUT).get());
+        final Connection databaseConnection = database.getConnection();
+        final String inputPath = this.optionAndArgumentDelegate.getOptionArgument(FLAG_PATH_INPUT)
+                .get();
+        this.timestamp = new Timestamp(Instant.now().toEpochMilli());
+
+        new File(inputPath).listFilesRecursively().parallelStream().forEach(file ->
```

Comment:
Why parallel stream here? And could this cause any read/write synchronization problems on the DB

Reply:
At this point, the db connection is already made and the only operations are writes; non of which depend on each other b/c there’s no table relationships.

parallel stream is used to speed up the file reading processing. I think I pulled it from the flag stats and mr upload command